src/SonarQube.TeamBuild.Integration/CoverageReportProcessorBase.cs

```diff
@@ -68,55 +67,50 @@ public bool ProcessCoverageReports()
 
             var success = TryGetBinaryReportFile(config, settings, logger, out string binaryFilePath);
 
-            if (success && binaryFilePath != null)
+            if (success &&
+                binaryFilePath != null &&
+                TryConvertCoverageReport(binaryFilePath, out var coverageReportPath) &&
+                !string.IsNullOrEmpty(coverageReportPath) &&
+                !config.LocalSettings.Any(IsVsCoverageXmlReportsPaths))
```

Comment:
Do we still want to perform the conversion if the user has supplied the path? (possibly yes, but it would be worth adding a comment as it's not obvious).